src/ngraph/runtime/intelgpu/intelgpu_op_custom_kernels.cpp

```diff
@@ -617,10 +617,20 @@ void runtime::intelgpu::do_logic_kernel(cldnn::topology& topology,
     const string entry_point_name = "logic_" + output_name;
     codegen::CodeWriter writer;
 
-    writer << "__kernel void " << entry_point_name << "(const __global float inputA"
-           << array_dims(inputA_shape) << ", const __global float inputB"
-           << array_dims(inputB_shape) << ", __global char output" << array_dims(output_shape)
-           << ")\n";
+    if (operation == " && " || operation == " || ")
```

Comment:
Please avoid string comparison as much as possible. This is not important here but it is better to be avoided.
I would like to ask you to redevelop/refactor the code a bit. As I can see you would like to use diffrent input parameters type for this kernel. I would propose to use `op->inputs().at(0).get_tensor().get_element_type().c_type_string()`  to get the type of input parameters in `do_logical_operation()`, add types to the function interface and use them in the proper line generation.
I expect it will generate `bool input[][]` instead `char` for boolean datatype including cases already covered by the function.
Please check tests for all affected operations before commit.